pkg/jsonschema/type.go

```diff
@@ -58,33 +58,33 @@ func getNullableNode(name string, typeInterface any, strict bool) (map[string]an
 	return node, nil
 }
 
-func getNodeFromSlice(in []any, strict bool) (map[string]any, error) {
+func getNodeFromSlice(in []any, options CreateSchemaOptions) (map[string]any, error) {
 	switch in[0] {
 	// "object" affects additionalProperties, properties, type and required
 	case "object":
-		return getObject(in, strict)
+		return getObject(in, options)
 	// "map" affects additionalProperties and type.
 	case "map":
-		return getMap(in, strict)
+		return getMap(in, options)
 	// "list" affects items, type
 	case "list":
-		return getList(in, strict)
+		return getList(in, options)
 	// "set" affects items, type, uniqueItems
 	case "set":
-		return getSet(in, strict)
+		return getSet(in, options)
 	// "tuple" affects items, type, maxItems, minItems
 	case "tuple":
-		return getTuple(in, strict)
+		return getTuple(in, options)
 	default:
 		panic("unknown type")
 	}
 }
 
-func getObject(in []any, strict bool) (map[string]any, error) {
+func getObject(in []any, options CreateSchemaOptions) (map[string]any, error) {
 	node := map[string]any{
 		"type": "object",
 	}
-	if strict {
+	if !options.AllowAdditionalProperties {
```

Comment:
Do we need the ! in the if statement? ie this may scan a little easier without the negation.

Reply:
kicking myself for this one now. In my defence there were 2 PRs for this line that i put up in different orders.